### PR TITLE
Move directory

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -319,15 +319,16 @@ private void moveElementDirectory(UUID elementUuid, UUID newDirectoryUuid, Strin
 
         DirectoryElementEntity oldDirectory = repositoryService.getElementEntity(element.getParentId()).orElseThrow();
         updateElementParentDirectory(element, newDirectoryUuid);
```
Comment (jamal-khey) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:321]: i think something is missing here.
when a directory is moved we need to find all the element in this directory and element in his sub dir and reindex the in elastic search.
you can check that the behavior is not correct by moving a dir that contains some elements , and making a search in gridexplore

**Review by SlimaneAmar — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Just need a test to move a root directory

**Comment by SlimaneAmar** [comment]
Remove Sonar issue !

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 2]
```diff
@@ -327,26 +328,43 @@ public void moveElementsDirectory(List<UUID> elementsUuids, UUID newDirectoryUui
...
+        elementsUuids.forEach(elementUuid -> moveElementDirectory(getDirectoryElementEntity(elementUuid), newDirectoryUuid, userId));
     }
 
-    private void moveElementDirectory(UUID elementUuid, UUID newDirectoryUuid, String userId) {
-        DirectoryElementEntity element = repositoryService.getElementEntity(elementUuid)
-                .orElseThrow(() -> DirectoryException.createElementNotFound(ELEMENT, elementUuid));
+    private void moveElementDirectory(DirectoryElementEntity element, UUID newDirectoryUuid, String userId) {
+        if (element.getParentId() == newDirectoryUuid) { // Same directory ?
+            return;
+        }
```
Comment (jamal-khey) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:337]: small bug here , '==' does not work on uuids
```suggestion
        if (element.getParentId().equals(newDirectoryUuid)) { // Same directory ?
            return;
        }
```

may be we need a small unit test for this case also ?

**Review by jamal-khey — COMMENTED** [review]
The tests and code look good overall. However, there is an issue with moving the directory to the same location, as we still receive a notification on the front end.

**Review by jamal-khey — APPROVED** [review]

**Review by souissimai — APPROVED** [review]

**Pull request merged**